src/hotspot/cpu/x86/assembler_x86.cpp

```diff
@@ -13007,18 +12939,106 @@ int Assembler::vex_prefix_and_encode(int dst_enc, int nds_enc, int src_enc, VexS
+  bool is_prefixq = (size == EVEX_64bit) ? true : false;
+  if (is_demotable(no_flags, dst_enc, nds_enc)) {
+    int encode = is_prefixq ? prefixq_and_encode(src_enc, dst_enc, is_map1) : prefix_and_encode(src_enc, dst_enc, is_map1);
+    emit_opcode_prefix_and_encoding((unsigned char)byte1, 0xC0, encode, imm8);
+  } else {
+    InstructionAttr attributes(AVX_128bit, is_prefixq, /* legacy_mode */ false, /* no_mask_reg */ true, /* uses_vl */ false);
+    attributes.set_address_attributes(/* tuple_type */ EVEX_NOSCALE, size);
+    int encode = evex_prefix_and_encode_ndd(src_enc, dst_enc, nds_enc, pre, opc, &attributes, no_flags);
+    emit_int24(byte1 & ~0x80, (0xC0 | encode), imm8);
+  }
+}
+
+void Assembler::evex_opcode_prefix_and_encode(int dst_enc, int nds_enc, int src_enc, VexSimdPrefix pre, VexOpcode opc,
+                                              int size, int byte1, bool no_flags, bool is_map1) {
+  bool is_prefixq = (size == EVEX_64bit);
+  if (is_demotable(no_flags, dst_enc, nds_enc)) {
+    if (size == EVEX_16bit) {
+      emit_int8(0x66);
+    }
+    int encode = is_prefixq ? prefixq_and_encode(src_enc, dst_enc, is_map1) : prefix_and_encode(src_enc, dst_enc, is_map1);
+    emit_opcode_prefix_and_encoding((unsigned char)byte1, 0xC0, encode);
+  } else {
+    InstructionAttr attributes(AVX_128bit, is_prefixq, /* legacy_mode */ false, /* no_mask_reg */ true, /* uses_vl */ false);
+    attributes.set_is_evex_instruction();
+    int encode = vex_prefix_and_encode(src_enc, dst_enc, nds_enc, pre, opc, &attributes, /* src_is_gpr */ true, /* nds_is_ndd */ true, no_flags);
+    emit_int16(byte1, (0xC0 | encode));
+  }
+}
+
+void Assembler::evex_opcode_prefix_and_encode_swap(int dst_enc, int nds_enc, int src_enc, VexSimdPrefix pre, VexOpcode opc,
+                                                   int size, int byte1, bool no_flags, bool is_map1) {
+  bool is_prefixq = (size == EVEX_64bit);
+  if (is_demotable(no_flags, dst_enc, nds_enc)) {
+    if (size == EVEX_16bit) {
+      emit_int8(0x66);
+    }
+    int encode = is_prefixq ? prefixq_and_encode(dst_enc, src_enc, is_map1) : prefix_and_encode(dst_enc, src_enc, is_map1);
+    emit_opcode_prefix_and_encoding((unsigned char)byte1, 0xC0, encode);
+  } else {
+    InstructionAttr attributes(AVX_128bit, is_prefixq, /* legacy_mode */ false, /* no_mask_reg */ true, /* uses_vl */ false);
+    attributes.set_is_evex_instruction();
+    int encode = vex_prefix_and_encode(nds_enc, dst_enc, src_enc, pre, opc, &attributes, /* src_is_gpr */ true, /* nds_is_ndd */ true, no_flags);
+    emit_int16(byte1, (0xC0 | encode));
+  }
+}
+
 int Assembler::evex_prefix_and_encode_ndd(int dst_enc, int nds_enc, int src_enc, VexSimdPrefix pre, VexOpcode opc,
-                           InstructionAttr *attributes, bool no_flags) {
+                                          InstructionAttr *attributes, bool no_flags, bool use_prefixq) {
```

Comment:
evex_prefix_and_encode_ndd => emit_eevex_prefix_or_demote_ndd

Naming suggestion.

Reply:
Changed as suggested in the updated code.